src/EditorFeatures/CSharpTest/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionTests.cs

```diff
@@ -973,6 +973,67 @@ public static void Test(string name)
             _ => null,
         };
     }
+}");
+        }
+
+        [WorkItem(40198, "https://github.com/dotnet/roslyn/issues/40198")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestNotWithRefReturns()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class Program
+{
+    static ref int GetRef(int[] mem, int addr, int mode)
+    {
+        [||]switch (mode)
+        {
+            case 0: return ref mem[mem[addr]];
+            case 1: return ref mem[addr];
+            default: throw new Exception();
+        }
+    }
+}");
+        }
+
+        [WorkItem(40198, "https://github.com/dotnet/roslyn/issues/40198")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestNotWithRefAssignment()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class Program
+{
+    static ref int GetRef(int[] mem, int addr, int mode)
+    {
+        ref int i = ref addr;
+        [||]switch (mode)
+        {
+            case 0: i = ref mem[mem[addr]]; break;
+            default: throw new Exception();
+        }
+
+        return ref mem[addr];
+    }
+}");
+        }
+
+        [WorkItem(40198, "https://github.com/dotnet/roslyn/issues/40198")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertSwitchStatementToExpression)]
+        public async Task TestNotWithRefConditionalAssignment()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class Program
+{
+    static ref int GetRef(int[] mem, int addr, int mode)
+    {
+        ref int i = ref addr;
+        [||]switch (mode)
+        {
+            case 0: i = true ? ref mem[mem[addr]] : ref mem[mem[addr]]; break;
```

Comment:
this is invalid code. ref is mandatory on ref-conditional.

Reply:
are you sure?  teh code compiled just fine for me.  try it out:

```c#
    static void GetRef(int[] mem, int addr, int mode)
    {
        ref int i = ref addr;
        switch (mode)
        {
            case 0: i = true ? ref mem[mem[addr]] : ref mem[mem[addr]]; break;
            case 1: i = ref mem[addr]; break;
            default: throw new Exception();
        }
    }
```

Reply:
That's right. but in those situations you can actually use switch expression so we don't need to worry about nested ref expressions.

Reply:
yup.  fixed up!